### PR TITLE
Adding Winston console logger to hubController class

### PR DESCRIPTION
**Pull request by shalinijoshi19**
(NOTE: I had accidentally merged earlier commit to master; reverted that; Re-Creating Pull Request)
- Updates package.json
- Updates HubController class to reference OpenT2T console logger.
- Default log levels is "info"
- Default log levels for testing is "verbose"

### CODE REVIEW

File: node/hubController.js [hunk 1]
```diff
@@ -3,15 +3,17 @@
...
-var hubsConfig = require("./hubsConfig");
+var HubsConfig = require("./hubsConfig");
 var OpenT2TError = require('opent2t').OpenT2TError;
 var OpenT2TConstants = require('opent2t').OpenT2TConstants;
+var OpenT2TLogger = require('opent2t').Logger;
 
 class HubController {
 
-    constructor() {
+    constructor(logLevel = "info") {
```
Comment (arjun-msft) [inline node/hubController.js]: >logLevel = "info" [](start = 16, length = 17)

why do we need to pass in a loglevel?

File: node/hubController.js [hunk 2]
```diff
@@ -3,15 +3,17 @@
...
         this.supportedHubsCache = undefined;
         this.OpenT2T = require('opent2t').OpenT2T;
+        this.ConsoleLogger = new OpenT2TLogger(logLevel);
```
Comment (arjun-msft) [inline node/hubController.js]: why is this named ConsoleLogger? won't it be whatever opent2tlogger is configured for?

File: node/hubController.js [hunk 3]
```diff
@@ -302,8 +301,8 @@ class HubController {
     }
     
     _setProperty(translatorName, deviceInfo, property, deviceId, value) {
-        console.log("----------------- _setProperty " + translatorName + " for " + property + " to:");
-        console.log(JSON.stringify(value, null, 2));
+        this.ConsoleLogger.verbose(
+            `_setProperty ${translatorName} for ${property} to ${JSON.stringify(alue, null, 2)}`);
```
Comment (arjun-msft) [inline node/hubController.js]: >alue [](start = 80, length = 4)

value not alue

File: node/hubController.js [hunk 4]
```diff
@@ -289,7 +287,8 @@ class HubController {
     }
 
     _invokeMethod(translator, deviceInfo, methodName, params) {
-        console.log("----------------- _invokeMethod " + methodName + " with params "+ JSON.stringify(params));
+        this.ConsoleLogger.verbose(
+            `_invokeMethod ${methodName} with params ${JSON.stringify(params)}`);
```
Comment (arjun-msft) [inline node/hubController.js]: I believe you need to do params, null, 2

File: node/hubController.js [hunk 5]
```diff
@@ -335,14 +334,14 @@ class HubController {
...
-        console.log(" custom error message:           " + customError.message);
-        console.log(" custom error name:           " + customError.name);
-        console.log(" custom error statusCode:           " + customError.statusCode);
-        console.log(" custom error innerError message:           " +  customError.innerError.message);
-        console.log(" custom error innerError stack:           " +  customError.innerError.stack);
+        this.ConsoleLogger.verbose(" custom error message:           " + customError.message);
+        this.ConsoleLogger.verbose(" custom error name:           " + customError.name);
+        this.ConsoleLogger.verbose(" custom error statusCode:           " + customError.statusCode);
+        this.ConsoleLogger.verbose(" custom error innerError message:           " + customError.innerError.message);
+        this.ConsoleLogger.verbose(" custom error innerError stack:           " + customError.innerError.stack);
```
Comment (arjun-msft) [inline node/hubController.js]: why not template values here?

**Comment by arjun-msft** [comment]
:clock1:

File: node/hubController.js (hunk 1 above)
Comment (shalinijoshi19) [inline node/hubController.js]: So we can control verbosity - say if you run in a diagnostic environment you might want to pass in "verbose", in prod you might "info" OR you might want to tune it down to "warn" or "error"

---
In reply to: [103827843](https://github.com/openT2T/cloud/pull/26#discussion_r103827843) [](ancestors = 103827843)

File: node/hubController.js (hunk 2 above)
Comment (shalinijoshi19) [inline node/hubController.js]: Yes but for now i wanted the name to be very clear around this logging to console only and we are instantiating opent2tlogger in default console mode only and not specifying a default file transport. (i didnt want to create random files in the cloud environement ) ; I expect the name to change and be more generic once we add support for the custom transport here.

---
In reply to: [103827940](https://github.com/openT2T/cloud/pull/26#discussion_r103827940) [](ancestors = 103827940)

File: node/hubController.js (hunk 5 above)
Comment (shalinijoshi19) [inline node/hubController.js]: missed. Fixed  now.

---
In reply to: [103828244](https://github.com/openT2T/cloud/pull/26#discussion_r103828244) [](ancestors = 103828244)

File: node/hubController.js (hunk 4 above)
Comment (shalinijoshi19) [inline node/hubController.js]: Ok so we arent consistent with this then? (To be clear this adds a space of 2 characters as whitespace? ) 

---
In reply to: [103828178](https://github.com/openT2T/cloud/pull/26#discussion_r103828178) [](ancestors = 103828178)

**Comment by danielrischar** [comment]
:shipit:

File: node/hubController.js (hunk 4 above)
Comment (arjun-msft) [inline node/hubController.js]: yes

---
In reply to: [103834273](https://github.com/openT2T/cloud/pull/26#discussion_r103834273) [](ancestors = 103834273,103828178)

**Comment by ChuckFerring** [comment]
:shipit:

**Pull request merged**